miner/worker.go

```diff
@@ -551,32 +529,33 @@ func (w *worker) taskLoop() {
 func (w *worker) resultLoop() {
 	for {
 		select {
-		case block := <-w.resultCh:
-			// Short circuit when receiving empty result.
-			if block == nil {
+		case result := <-w.resultCh:
+			// Short circuit when receiving empty result
+			if result == nil {
 				continue
 			}
+			block := result.SealedBlock
 			// Short circuit when receiving duplicate result caused by resubmitting.
 			if w.chain.HasBlock(block.Hash(), block.NumberU64()) {
 				continue
 			}
 			var (
 				sealhash = w.engine.SealHash(block.Header())
 				hash     = block.Hash()
+				r        = result.BlockProcessResult
```

Comment:
nit:  Can this variable be renamed to something more informative.  e.g. `proccessResult`.